src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -220,9 +220,16 @@ public override BoundStatement InstrumentSwitchStatement(BoundSwitchStatement or
             return AddDynamicAnalysis(original, base.InstrumentSwitchStatement(original, rewritten));
         }
 
-        public override BoundStatement InstrumentBoundPatternSwitchStatement(BoundPatternSwitchStatement original, BoundStatement rewritten)
+        public override BoundStatement InstrumentPatternSwitchStatement(BoundPatternSwitchStatement original, BoundStatement rewritten)
         {
-            return AddDynamicAnalysis(original, base.InstrumentBoundPatternSwitchStatement(original, rewritten));
+            return AddDynamicAnalysis(original, base.InstrumentPatternSwitchStatement(original, rewritten));
+        }
+
+        public override BoundStatement InstrumentPatternSwitchWhenClause(BoundExpression original, BoundStatement rewritten)
+        {
+            // TODO: Desired behavior for a switch when clause has not been specified.
+            // See also https://github.com/dotnet/roslyn/issues/14156
+            return base.InstrumentPatternSwitchWhenClause(original, rewritten);
```

Comment:
Deciding how to instrument this construct (aside from sequence points for debugging) is probably a larger discussion. I looked at `DynamicAnalysisInjector` but I'd rather treat instrumentation for this as a separate work item.